src/plugins/telemetry_collection_manager/server/plugin.ts

```diff
@@ -309,36 +309,30 @@ export class TelemetryCollectionManagerPlugin
     const clustersDetails = await collection.clusterDetailsGetter(statsCollectionConfig, context);
     const { refreshCache } = statsCollectionConfig;
     const { title: collectionSource } = collection;
-    this.cacheManager.unrefExpiredCacheObjects();
 
     // on `refreshCache: true` clear all cache to store a fresh copy
     if (refreshCache) {
-      this.cacheManager.unrefAllCacheObjects();
+      this.cacheManager.resetCache();
     }
 
     if (clustersDetails.length === 0) {
       return [];
     }
 
-    const cacheKey = this.createCacheKey(collection, clustersDetails);
-    if (this.cacheManager.isCacheValid(cacheKey)) {
-      const cachedData = this.cacheManager.getFromCache<BasicStatsPayload[]>(cacheKey);
-      if (cachedData) {
-        const { data, cacheTimestamp } = cachedData;
-        return data.map((stat) => ({
-          cacheDetails: {
-            isCached: true,
-            cacheTimestamp,
-          },
-          collectionSource,
-          ...stat,
-        }));
-      }
+    const cacheKey = this.createCacheKey(collectionSource, clustersDetails);
+    const cachedUsageStatsPayload = this.cacheManager.getFromCache<UsageStatsPayload[]>(cacheKey);
+    if (cachedUsageStatsPayload) {
+      return cachedUsageStatsPayload;
     }
 
     const stats = await collection.statsGetter(clustersDetails, statsCollectionConfig, context);
-    this.cacheManager.setCache(cacheKey, stats);
+    const usageStatsPayload = stats.map((stat) => ({
+      collectionSource,
+      cacheDetails: { updatedAt: new Date().toISOString() },
```

Comment:
Q: AFAIK, `stats` contain the `timestamp`, and we are storing it in the cache. Will `cacheDetails.updatedAt` and `timestamp` always be around the same time? How are we planning to use this field?

Reply:
True actually. The main use case I can think of is allowing debuggability. We can drop the entire field then

Reply:
After an async discussion we are missing some detail here which is the `timestamp` the data was fetched from cache since the `timestamp` in the `stats` is cached with the payload. We'll keep the `updatedAt` as well to help debugging and keep concerns separate. These cache details are needed to aid in debugging and understand the caching mechanism better as time goes on.
1. I'll add `cacheDetails.fetchedAt` which is updated at the time of fetch everytime
2. I'll add documentation for these fields